docs/06-ReusingTestCode.md

```diff
@@ -0,0 +1,636 @@
+$I->login('miles', '123456');
+?>
+```
+
+However, implementing all actions for a reuse in one actor class may lead to breaking the [Single Responsibility Principle](http://en.wikipedia.org/wiki/Single_responsibility_principle). 
+
+## StepObjects
+
+If `login` method defined in Actor class may be used in 90% of your tests,
+StepObjects are great if you need some common functionality for a group of tests. Let's say you are going to test and admin area of a site. Probably you won't need the same actions from admin area while testing the frontend, so it's a good idea to move those admin-specific into their own class. We will call such class a StepObject.
+
+Lets create a Adnin StepObject with generator, by specifying test suite, and passing method expected names on prompt.
+
+```bash
+$ php codecept.phar generate:stepobject acceptance Admin
+```
+
+You will be asked to enter action names, but it's optional. Enter one at a time, and press Enter. After specifying all needed actions, leave empty line to go on to StepObject creation.
+
+```bash
+$ php codecept.phar generate:stepobject acceptance Admin
+Add action to StepObject class (ENTER to exit): loginAsAdmin
+Add action to StepObject class (ENTER to exit):
+StepObject was created in /tests/acceptance/_support/Step/Acceptance/Admin.php
+```
+
+It will generate class in `/tests/_support/Step/Acceptance/Admin.php` similar to this:
+
+```php
+<?php
+namespace Step\Acceptance;
+
+class Admin extends \AcceptanceTester
+{
+    public function loginAsAdmin()
+    {
+        $I = $this;
+    }
+}
+?>
+```
+
+As you see, this class is very simple. It extends `AcceptanceTester` class, thus, all methods and properties of `AcceptanceTester` are available for usage in it.
+
+`loginAsAdmin` method may be implemented like this:
+
+```php
+<?php
+namespace Step\Acceptance;
+
+class Member extends \AcceptanceTester
+{
+    public function loginAsAdmin()
+    {
+        $I = $this;
+        $I->amOnPage('/admin');
+        $I->fillField('username', 'admin');
+        $I->fillField('password', '123456');
+        $I->click('Login');
+    }
+}
+?>
+```
+
+In tests you can use a StepObject by instantiating `Step\Acceptance\Admin` instead of `AcceptanceTester`.
+
+```php
+<?php
+use Step/Acceptance/Admin as AdminTester;
+
+$I = new AdminTester($scenario);
+$I->loginAsAdmin();
+?>
+```
+
+Same as above, StepObject can be instanticated automatically by Dependency Injection Container, when used inside Cest format:
+
+```php
+<?php
+class UserCest 
+{    
+    function showUserProfile(\Step\Acceptance\Admin $I)
+    {
+        $I->loginAsAdmin();
+        $I->amOnPage('/admin/profile');
+        $I->see('Admin Profile', 'h1');        
+    }
+}
+?>
+```
+
+If you have complex interaction scenario you may use several page objects in one test. If you feel like adding too much actions into your Actor class (which is AcceptanceTester in this case) consider to move some of them into separate StepObjects.
```

Comment:
@javigomez 

> Could you advise me how to do the autoloading part? should we use the _bootstrap file?

If you use Composer autoloading should be handled by it. If not, you can rely on bootstrap file. If you wrap StepObjects into package you can install them and use in Codeception tests.

In case of Joomla, WP, Drupal tests, I'd recommend to have PageObjects with all common locators defined. They may suit better than StepObjects, as you can take a particular locator of a page and use it in the test, as well as you can define actions which will be used there.